### ISSUE TITLE
Set up custom domain/CDN for FECFile API, update app settings with new domain

### ISSUE THREAD

**Issue by lbeaufort** [fecgov/fecfile-web-api#780]
Business Reason

As a technical requirement, the FECfile API should be on a fec.gov domain and not cloud.gov domain.

As a user, I want to trust that I'm on a FEC system and the URL [https://fecfile-web-api-prod.app.cloud.gov/](https://fecfile-web-api-prod.app.cloud.gov/) isn't on the FEC domain.

In order for the lower environments, we should set this up in dev/stage as well. Something like:

*   dev-api.fecfile.fec.gov
*   stage-api.fecfile.fec.gov
*   api.fecfile.fec.gov

See also [#1493](https://github.com/fecgov/fecfile-web-app/issues/1493)

### QA Notes ###

 
  
 - [ ] BA/Design Quality Checklist complete

### DEV Notes ###

 
  
 - Use AWS Route 53 to set up DNS records - FEC DNS is a little wonky so it will need to be an apex domain. So it might make the most sense to make fecfile.fec.gov an apex and do subdomains for dev.fecfile.fec.gov, stage.fecfile.fec.gov.
- If there is a difference between the app and api domains (currently, they are fecfile-web-api-stage.app.cloud.gov and fecfile-web-app-stage.app.cloud.gov for example), then the FFAPI_COOKIE_DOMAIN environment variable will need to be set to a value that includes both (currently, it is set to .app.cloud.gov for example) so that the client can access necessary api cookies.
- Also need to update `CORS_ALLOWED_ORIGINS` and `CSRF_TRUSTED_ORIGINS` - should it be any fec.gov?
- Can't follow https://cloud.gov/docs/services/external-domain-service/ exactly because AWS Route 53 won't let us set up an A or CNAME for an apex (`fecfile-dev-api.fec.gov`) that points to anything other than an AWS resource. Need to set up CDN service and then ask cloud.gov for cloudfront distribution: `cf create-service external-domain domain-with-cdn fecfile-dev-api.fec.gov -c '{"domains": "fecfile-dev-api.fec.gov"}'` More 

notes: 🔒 https://docs.google.com/document/d/1zFLq3Y-VmUpXOI0cXq_wIhH0ybdgua07s5vrosd6z3o/edit 🔒 

domain-with-cdn plan: https://cloud.gov/docs/services/external-domain-service/

**PRs:** 
https://github.com/fecgov/fecfile-api-proxy/pull/7
https://github.com/fecgov/fecfile-web-api/pull/945
https://github.com/fecgov/fecfile-web-app/pull/2019

- [ ] Remove/unmap old proxy api routes once they're working properly
- [x] Update login.gov redirect URLs in partner dashboard
- [ ] Communication strategy - share new routes with the team

### Design ###

 
  
 null

[FECFILE-206](https://fecgov.atlassian.net/browse/FECFILE-206)

**Comment by MitchellTCG** [comment]
@lbeaufort will add a related ticket about redirects

**Comment by AureliaKhorsand** [comment]
@MitchellTCG @lbeaufort why is this on hold? Can we add that to a comment in this ticket?

**Comment by lbeaufort** [comment]
@AureliaKhorsand this ticket is blocked by https://github.com/fecgov/fecfile-web-api/issues/816. Once #816 is done, we can pick up this ticket.

**Comment by lbeaufort** [comment]
While #816 isn't merged, I have enough to test in dev, so removed the "on hold" label.

**Comment by mjtravers** [comment]
Passes CR. Sending to QA.

All browsers are able to use the new URL at https://dev.fecfile.fec.gov/login/
